c2corg_ui/static/js/addassociation.js

```diff
@@ -88,7 +88,8 @@ app.AddAssociationController.prototype.associate = function(doc) {
   // if the parent doc is a route and the child doc is a waypoint OR
   // if the parent doc is an outing, inverse the IDs.
   if ((parentType === 'routes' && doc['type'] === 'w') ||
-      parentType === 'outings' || parentType === 'images' ||
+      (parentType === 'outings' && (doc['type'] === 'r' || doc['type'] === 'u')) ||
```

Comment:
OK thanks for the explanations.

To be honest I think this part is getting a bit confusing and not easy to understand. I don't know if we could do something to make it more explicit?